### PR TITLE
Relaxed clippy fixes/nits

### PR DESCRIPTION
**Pull request by bkontur**
This PR contains just a few clippy fixes and nits, which are, however, relaxed by workspace clippy settings here: https://github.com/paritytech/polkadot-sdk/blob/master/Cargo.toml#L483-L506

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4664434

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4667144

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4667413

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4667685

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4668350

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4668497

**Review by koute — APPROVED** [review]

File: .gitlab/pipeline/check.yml [hunk 1]
```diff
@@ -4,8 +4,11 @@ cargo-clippy:
     - .docker-env
     - .common-refs
     - .pipeline-stopper-artifacts
+  variables:
+    RUSTFLAGS: "-Dwarnings"
   script:
-    - SKIP_WASM_BUILD=1 env -u RUSTFLAGS cargo clippy --all-targets --locked --workspace
+    # TODO: remove before merge - temporary allow exceptions
+    - SKIP_WASM_BUILD=1 cargo clippy --all-targets --locked --workspace -- --allow clippy::type_complexity
```
Comment (ggwpez) [inline .gitlab/pipeline/check.yml:11]:
```suggestion
```

**Comment by bkontur** [comment]
now I/we know how clippy works, those `-A` / `-D` from [`.cargo/config.toml`](https://github.com/paritytech/polkadot-sdk/blob/master/.cargo/config.toml#L11-L12) are added automatically, so yes, that removal `env -u RUSTFLAGS` is really needed.

I would suggest to merge https://github.com/paritytech/polkadot-sdk/pull/2669 at first.
And then we can decide if we want to merge also **code changes** from this PR (after reverting CI files)

File: .gitlab/pipeline/check.yml (hunk 1 above)
Comment (liamaharon) [reply .gitlab/pipeline/check.yml:11]: I think you mean just remove `--allow clippy::type_complexity` @ggwpez ?

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4712271

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4712810

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4713991

File: polkadot/node/network/bridge/src/tx/mod.rs [hunk 2]
```diff
@@ -30,7 +30,6 @@ use polkadot_node_subsystem::{
 /// Peer set info for network initialization.
 ///
 /// To be passed to [`FullNetworkConfiguration::add_notification_protocol`]().
```
Comment (ordian) [inline polkadot/node/network/bridge/src/tx/mod.rs:32]:
```suggestion
```

**Comment by bkchr** [comment]
@bkontur what is the status of this?

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4737708

**Comment by bkontur** [comment]
bot fmt

**Comment by bkontur** [comment]
> @bkontur what is the status of this?

@bkchr it is ready now, it just contains @s0me0ne-unkn0wn's clippy nits (and few mine), which are relax by workspace clippy config, so I guess no problem to merge

**Review by ordian — APPROVED** [review]

File: substrate/frame/contracts/src/wasm/mod.rs [hunk 3]
```diff
@@ -25,13 +25,15 @@ mod runtime;
...
-	RuntimeCosts,
+	AllowDeprecatedInterface, AllowUnstableInterface, Environment, Runtime, RuntimeCosts,
 };
-pub use pallet_contracts_uapi::ReturnFlags;
+
 #[cfg(test)]
 pub use tests::MockExt;
 
+#[cfg(test)]
+pub use crate::wasm::runtime::ReturnErrorCode;
```
Comment (bkchr) [inline substrate/frame/contracts/src/wasm/mod.rs:35]:
```suggestion
use crate::wasm::runtime::ReturnErrorCode;
```

File: substrate/frame/fast-unstake/src/lib.rs [hunk 4]
```diff
@@ -571,7 +571,7 @@ pub mod pallet {
 					.any(|e| T::Staking::is_exposed_in_era(&stash, e));
 
 				if is_exposed {
-					T::Currency::slash_reserved(&stash, deposit);
+					let (_, _) = T::Currency::slash_reserved(&stash, deposit);
```
Comment (bkchr) [inline substrate/frame/fast-unstake/src/lib.rs:574]:
```suggestion
					let _ = T::Currency::slash_reserved(&stash, deposit);
```

File: substrate/frame/indices/src/lib.rs [hunk 5]
```diff
@@ -223,7 +223,7 @@ pub mod pallet {
 				let (account, amount, perm) = maybe_value.take().ok_or(Error::<T>::NotAssigned)?;
 				ensure!(!perm, Error::<T>::Permanent);
 				ensure!(account == who, Error::<T>::NotOwner);
-				T::Currency::slash_reserved(&who, amount);
+				let (_, _) = T::Currency::slash_reserved(&who, amount);
```
Comment (bkchr) [inline substrate/frame/indices/src/lib.rs:226]:
```suggestion
				let _ = T::Currency::slash_reserved(&who, amount);
```

**Review by bkchr — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4742946

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4742962

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4742964

File: substrate/frame/contracts/src/wasm/mod.rs [hunk 6]
```diff
@@ -25,13 +25,15 @@ mod runtime;
...
-	RuntimeCosts,
+	AllowDeprecatedInterface, AllowUnstableInterface, Environment, Runtime, RuntimeCosts,
 };
-pub use pallet_contracts_uapi::ReturnFlags;
+
 #[cfg(test)]
 pub use tests::MockExt;
 
+#[cfg(test)]
+use crate::wasm::runtime::ReturnErrorCode;
```
Comment (bkchr) [inline substrate/frame/contracts/src/wasm/mod.rs:35]:
```suggestion
pub use crate::wasm::runtime::ReturnErrorCode;
```

**Pull request merged**